rust/relay/src/server.rs

```diff
@@ -727,24 +720,24 @@ where
         message: ChannelData,
         sender: ClientSocket,
         _: SystemTime,
-    ) {
-        let channel_number = message.channel();
+    ) -> Option<(AllocationPort, PeerSocket)> {
+        let channel_number = message.channel().value();
         let data = message.data();
 
         let Some(channel) = self
             .channels_by_client_and_number
             .get(&(sender, channel_number))
         else {
             tracing::debug!(target: "relay", channel = %channel_number, "Channel does not exist, refusing to forward data");
-            return;
+            return None;
         };
 
         // TODO: Do we need to enforce that only the creator of the channel can relay data?
         // The sender of a UDP packet can be spoofed, so why would we bother?
```

Comment:
That reminds me, how does auth work once a channel is open? Is there some cryptography so that a third party can't send anything to a relay port and have it relayed?

Or is it just based on IP:port tuple and if the Client roams they reconnect using the network change logic? Hence #4267

Reply:
There is no auth for channels. You can spoof your IP and port and send traffic to it. But all the traffic is WG so you can't actually do any harm.